File: Homework2/FTI/FTI.Api/Models/Receipt.cs
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json;

namespace FTI.Api.Models
{
    public class Receipt
    {

        public Receipt()
        {
            this.Items = new List<Item>();
        }

        public string CustomerNumber { get; private set; }

        public IList<Item> Items { get; }

        public Amount Total
        {
            get { return new Amount(CurrencyEnum.EUR, Items.Sum(x => x.Price.Value)); }
        }

        public void AddItem(Item item)
        {
            this.Items.Add(item);
        }

        public void AddCustomerNumber(string customerNumber)
        {
            this.CustomerNumber = customerNumber;
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this);
        }
    }
}

File: Homework2/FTI-WebClient/src/models/Receipt.ts
import { Item } from './Item';
import { Amount } from './Amount';

export class Receipt {  
  CustomerNumber: string;
  Items: Item[] = [];
  Total: Amount;
  
  constructor() {
      this.Total = new Amount("EUR", 0);
  }
  
  get total():Amount{
    var sum = this.Items.reduce((sum, current) => sum + current.Price.Value, 0);

    return new Amount("EUR", sum);
  }
}